tests/apollo/test_skvbc_restart_recovery.py

```diff
@@ -470,6 +470,56 @@ async def test_view_change_with_non_primary_replica_in_state_transfer(self, bft_
                 run_ops=lambda: skvbc.run_concurrent_ops(num_ops=20, write_weight=1), threshold=20)
             log.log_message("fast path prevailed")
 
+    @skip_for_tls
+    @with_trio
+    @with_bft_network(start_replica_cmd)
+    @verify_linearizability(pre_exec_enabled=True, no_conflicts=True)
+    @with_constant_load
+    async def test_recovering_view_after_restart_with_packet_loss(self, bft_network, skvbc, tracker, constant_load):
+        """
+        Implement a test with constant client load that is running under the Packet Dropping Adversary
+        with 50% chance of dropping a packet from all replicas and trigger View Changes periodically.
+        The test should run for 10 minutes and in the end we drop the adversarial packet dropping
+        activity and verify that we will regain Fast Commit path in the system.
+        Step by step scenario:
+        1. Create a packet dropping adversary with 50% chance of dropping any packet.
+        2. Introduce constant client load.
+        3. Restart the current primary to trigger View Change.
+        4. Wait for View Change to complete (because of the packet dropping activity this could take significantly more time than usual and a non-deterministic view increments can happen).
+        5. Loop to step 3.
+        """
+        # uncomment for live tracking of log messages from the test
+        # class foo:
+        #     def log_message(self, var):
+        #         print(f"{var}")
+
+        # log = foo()
+        # start replicas
+        [bft_network.start_replica(i) for i in bft_network.all_replicas()]
+
+        loop_count_outer = 0
+
+        while (loop_count_outer < 20):
+            loop_count_outer = loop_count_outer + 1
+            loop_count = 0
+            primary = 0
+            while (loop_count < 5):
+                loop_count = loop_count + 1
+                log.log_message(f"Loop run: {loop_count}")
+                with net.PacketDroppingAdversary(bft_network, drop_rate_percentage=50) as adversary:
+                    adversary.interfere()
+
+                    primary = await bft_network.get_current_primary()
```

Comment:
I would suggest to move the getting of the primary before we enter the section with the packet dropping adversary. This way we will get it on a stable point and will restart it right after we enter the scope with the packet dropping adversary to make sure View Change happens with certainty.